### PR TITLE
IntoChecked Predicate Verification

### PR DESCRIPTION
**Pull request by Voxelot**
Finish todo for making IntoChecked perform full verification including predicates

### CODE REVIEW

File: fuel-vm/src/checked_transaction.rs [hunk 1]
```diff
@@ -110,27 +110,14 @@ impl<Tx: IntoChecked> Checked<Tx> {
     }
 }
 
-impl<Tx: IntoChecked + ExecutableTransaction + fuel_tx::field::GasLimit> Checked<Tx>
+// #[cfg(feature = "internals")]
```
Comment (Dentosal) [inline fuel-vm/src/checked_transaction.rs:113]: Should we just remove these instead of commenting out?

**Review by Dentosal — APPROVED** [review]
LGTM

**Review by mitch-fuel — APPROVED** [review]
Agreed, looks good

File: fuel-vm/src/checked_transaction.rs (hunk 1 above)
Comment (Voxelot) [reply fuel-vm/src/checked_transaction.rs:113]: I do it after this PR is merged so it doesn't undo the approvals

**Pull request merged**